tube/etl/outputs/es/timestamp.py

```diff
@@ -0,0 +1,58 @@
+import re
+from datetime import datetime
+import calendar
+from psycopg2.sql import SQL, Identifier
+from tube.utils.db import execute_sql_query_return_first_item
+
+
+def get_backup_version(index_name):
+    res = re.match('^[0-9]+', index_name)
+    if res is not None:
+        return int(res.group()) + 1
+    return 0
+
+
+def to_utc_time(dt):
+    utc_time = calendar.timegm(dt.utctimetuple()) + dt.microsecond / 1E6
+    return datetime.utcfromtimestamp(utc_time)
+
+
+def get_latest_utc_transaction_time():
```

Comment:
Isn't it would be better to use proper SQL string composition, instead of Python string interpolation?

```
fields = ["id", "state", "created_datetime"]
order_by_field = "created_datetime"
select_clause = SQL("SELECT {fields}").format(fields=SQL(",").join(map(Identifier, fields)))
from_clause = SQL("FROM {table}").format(table=Identifier("transaction_logs"))
where_clause = SQL("WHERE state='SUCCEEDED' ORDER BY {order_by_field} DESC").format(order_by_field=Identifier(order_by_field))
limit_clause = SQL("LIMIT 1")
query = SQL(" ").join([select_clause, from_clause, where_clause, limit_clause, SQL(";")])
```

Reply:
thanks, will fix it